### PR TITLE
Add extra properties for test results in dotnet test

### PR DESCRIPTION
**Pull request by mariam-abdulla**
This pull request introduces several enhancements to the test result messaging system, primarily focusing on adding new fields to test result messages and updating related serializers and handlers. The most important changes include extending the `SuccessfulTestResultMessage` and `FailedTestResultMessage` records, updating field IDs, and modifying the serializers to handle the new fields.

### Enhancements to Test Result Messages:

* [`src/Cli/dotnet/commands/dotnet-test/IPC/Models/TestResultMessages.cs`](diffhunk://#diff-80a8ee64c5a93a579172e4a8029381e412d207bbf076e5d23971d98d142c7ccbL1-R10): Added `Duration`, `StandardOutput`, and `ErrorOutput` fields to `SuccessfulTestResultMessage` and `FailedTestResultMessage` records.

### Updates to Field IDs:

* [`src/Cli/dotnet/commands/dotnet-test/IPC/ObjectFieldIds.cs`](diffhunk://#diff-ea80e935394c39a82e357278fbe17f67c9fe440373662e5d1a940265d7888d86L74-R92): Updated field IDs to include `Duration`, `StandardOutput`, and `ErrorOutput` for both `SuccessfulTestResultMessage` and `FailedTestResultMessage`.

### Serializer Modifications:

* [`src/Cli/dotnet/commands/dotnet-test/IPC/Serializers/BaseSerializer.cs`](diffhunk://#diff-b6eb529440a6cd2a0939dc396f83f71dce504bd821a0aae7e4ac1638f5537ee4R282-R293): Added a new method to write `long?` values to the stream.
* [`src/Cli/dotnet/commands/dotnet-test/IPC/Serializers/TestResultMessagesSerializer.cs`](diffhunk://#diff-4c5b4b6340b221fcf0c944ee4bff0dfd8ca585ccdce014c33cf79cbbb26f7343L35-R53): Updated the serializer to handle the new fields (`Duration`, `StandardOutput`, `ErrorOutput`) for both successful and failed test messages.

### Additional Changes:

* [`src/Cli/dotnet/commands/dotnet-test/Models.cs`](diffhunk://#diff-3ee56fe6c4fb8020f4ca49c84e9d3b4e858f3d62d307244d53669d94db84013fL16-R18): Added new fields to the `SuccessfulTestResult` and `FailedTestResult` records to match the changes in `TestResultMessages`.
* [`src/Cli/dotnet/commands/dotnet-test/TestApplication.cs`](diffhunk://#diff-87af23dcd9d529167e3c71f60d8f5077ea0eca3680f5b4a8c959e81a98701baaL338-R341): Updated the `OnTestResultMessages` method to handle the new fields in the test result messages.

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-test/IPC/Models/TestResultMessages.cs [hunk 1]
```diff
@@ -1,13 +1,13 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (nohwnd) [inline src/Cli/dotnet/commands/dotnet-test/IPC/Models/TestResultMessages.cs:1]: Why change the license?

Comment (mariam-abdulla) [reply src/Cli/dotnet/commands/dotnet-test/IPC/Models/TestResultMessages.cs]: This was done automatically upon saving. I removed this change.

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
This PR requires this change  on testfx to be merged: https://github.com/microsoft/testfx/pull/3932/files

**Review by nohwnd — APPROVED** [review]

**Comment by Evangelink** [comment]
Relates to #45927

**Pull request merged**